src/structs.jl

```diff
@@ -50,3 +50,173 @@ IPUMS.IPUMSSource("IPUMS USA", "usa", "microdata", "", true, "")
+strings is a space `" "`, while the default value for missing integer values 
+is 9999.
+
+
+# Keyword Arguments
+
+- `name::String` - Name of the variable, as per the column name of the IPUMs 
+                    extract file. This name is limited to 8 characters.
+- `position_start::Int64` - The starting position (in columns) of a variable 
+                            in a fixed width file format.
+- `position_end::Int64` - The ending position (in columns) of a variable 
+                            in a fixed width file format.
+- `position_width::Int64` - The length (in columns) of a variable in a fixed 
+                            width file format. 
+- `labl::String` - A short description of the variable. Often the `labl` is 
+                    used to display a description of the variable in a 
+                    dataframe or display.
+- `desc::String` - A longer description of the variable, including information 
+                    about the use of the variable. 
+- `dcml::Int64` - Identifies the number of decimal points in the variable.
+- `vartype::String` - Indentifies whether the variable is a string type or numeric
+                    type.
+- `varinterval::String` - Identifies if a numeric variable is discrete or 
+                    continuous. 
+
+# Returns
+
+- `DDIVariable` object specifying the metadata for each variable.
+
+# Example
+
+```julia-repl
+julia> DDIVariable(
+    name = "YEAR",
+    position_start = 1,
+    position_end = 4,
+    position_width = 4, 
+    labl = "Survey year",
+    desc = "YEAR reports the year in which the survey was conducted.  YEARP is repeated on person records.",
+    dcml = 0,
+    vartype = "numeric",
+    varinterval = "discrete"
+    )
+
+IPUMS.DDIVariable("YEAR", 1, 4, 4, "Survey year", "YEAR reports the year in which the survey was conducted.  YEARP is repeated on person records.", 0, "numeric", "discrete")
+DDIVariable("IPUMS USA", "usa", "microdata", "", true, "")
+```
+
+# References
+
+Information about each variable field is taken from: 
+
+https://ddialliance.org/Specification/DDI-Codebook/2.5/XMLSchema/field_level_documentation_files/schemas/codebook_xsd/elements/stdyDscr.html
+"""
+@kwdef mutable struct DDIVariable
+    name::String = ""
+    position_start::Int64 = 9999
+    position_end::Int64 = 9999
+    position_width::Int64 = 9999 
+    labl::String = ""
+    desc::String = ""
+    dcml::Int64 = 9999
+    vartype::String = ""
+    varinterval::String = ""
+end
+
+
+"""
+```julia
+DDIInfo(
+    filepath::String, 
+    conditions::String = "", 
+    citation::String = "", 
+    ipums_project::String = "",
+    extract_notes::Sring = "", 
+    extract_date::String = "",
+    variable_info::Vector{DDIVariable} = DDIVariable[]
+    _xml_doc::EzXML.Document = EzXML.XMLDocument()
+    _ns::String = ""
+)
+```
+
+A struct representing the metadata taken from an IPUMs extract. An IPUMs 
+extract contains both file-level metadata--such as the date of export--as well
```

Comment:
![image](https://github.com/JuliaHealth/IPUMS.jl/assets/29561456/d985e2eb-7e0d-4cfe-a053-261770fed652)

For some reason, the double dashes look odd here. Might be a REPL markdown rendering issue.

Reply:
I have now escaped the double dashes, as per https://stackoverflow.com/questions/16871088/how-to-escape-double-dash-in-markdown . Let's see if that solves the weirdness.